### PR TITLE
[Fire Mage] Blaster Master Trait

### PR DESCRIPTION
**Pull request by Sharrq**
Added support for the Blaster Master Trait. Keeps track of the max number of stacks you got to during Combustion.

Suggestion
image

Timeline
image

Stat Box
image

### CODE REVIEW

File: src/parser/mage/fire/CHANGELOG.js [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
 
 export default [
+  {
+    date: new Date('2018-11-15'),
+    changes: <>Added support for <SpellLink id={SPELLS.BLASTER_MASTER.id} />.</>,
+    contributors: [Sharrq],
+  },
+  {
```
Comment (Dambroda) [inline src/parser/mage/fire/CHANGELOG.js]: Something wrong with the diff here, you have two changelog entries for the hot streak changes

File: src/parser/mage/fire/modules/traits/BlasterMaster.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  statistic() {
+    return (
+      <TraitStatisticBox
+        position={STATISTIC_ORDER.OPTIONAL()}
+        trait={SPELLS.BLASTER_MASTER.id}
+        value={(<>{this.averageStacksPerCombustion.toFixed(2)} stacks per Combustion<br /></>)}
+        tooltip={
+          `Blaster Master is a somewhat complicated trait to get the full effect from and may involve adjusting your rotation during Combustion. In order to get the most out of this trait, you shoud aim to get to 4 stacks of the Blaster Master during Combustion. For additional information on how to adjust your rotation to accomplish this, refer to the Mage Discord or the link in the Suggestion.`
```
Comment (Dambroda) [inline src/parser/mage/fire/modules/traits/BlasterMaster.js]: > Blaster Master is a somewhat complicated trait to get the full effect from and may involve adjusting your rotation during Combustion. In order to get the most out of this trait, you **_shoud_** aim to get to 4 stacks of **_the_** Blaster Master during Combustion. For additional information on how to **_adjust your rotation_** to accomplish this, refer to the Mage Discord or the link in the Suggestion.

Few typos and such; suggested change:

> Blaster Master is a somewhat complicated trait and utilizing it to its full potential involves adjusting your rotation during Combustion. In order to get the most out of this trait, you should aim to get to 4 stacks of the Blaster Master buff during each of your Combustions. For a guide on how to accomplish this, refer to the Mage Discord or the link in the <i>Suggestions</i> tab above.

File: src/parser/mage/fire/modules/traits/BlasterMaster.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          .recommended(`${formatNumber(recommended)} is recommended`);
+      });
+  }
+
+  statistic() {
+    return (
+      <TraitStatisticBox
+        position={STATISTIC_ORDER.OPTIONAL()}
+        trait={SPELLS.BLASTER_MASTER.id}
+        value={(<>{this.averageStacksPerCombustion.toFixed(2)} stacks per Combustion<br /></>)}
```
Comment (Dambroda) [inline src/parser/mage/fire/modules/traits/BlasterMaster.js]: No need for the fragment here
```
value={`${this.averageStacksPerCombustion.toFixed(2)} stacks per Combustion`}
```

File: src/parser/mage/fire/modules/traits/BlasterMaster.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  badCombustion = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.BLASTER_MASTER.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.COMBUSTION), this.onCombustionCast);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.COMBUSTION), this.onCombustionStart);
+    this.addEventListener(Events.removebuff.to(SELECTED_PLAYER).spell(SPELLS.COMBUSTION), this.onCombustionEnd);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.BLASTER_MASTER_BUFF), this.onTraitStack);
+    this.addEventListener(Events.applybuffstack.to(SELECTED_PLAYER).spell(SPELLS.BLASTER_MASTER_BUFF), this.onTraitStack);
```
Comment (Dambroda) [inline src/parser/mage/fire/modules/traits/BlasterMaster.js]: I personally find this beautiful 👍

**Comment by Sharrq** [comment]
made the requested changes, though i made them from the github web ui, so im not sure if that is going to work the way i thought it would

**Pull request merged**